src/main/java/seedu/address/logic/LogicManager.java

```diff
@@ -33,8 +37,14 @@ public LogicManager(Model model, Storage storage) {
     public CommandResult execute(String commandText) {
         logger.info("----------------[USER COMMAND][" + commandText + "]");
         Command command = parser.parseCommand(commandText);
+
         command.setData(model);
-        return command.execute();
+        CommandResult result = command.execute();
+
+        if (model.hasUncommittedChanges() && !(command instanceof UndoCommand) && !(command instanceof RedoCommand) ) {
```

Comment:
its part of the implementation, because Undo and Redo do not add commits, so recordState should not be called with undo or redo. Undo or Redo only moves the head label back and forth the commit list